tests/conftest.py

```diff
@@ -41,14 +49,37 @@ def pytest_sessionstart(session):
     info = _get_info(redis_url)
     version = info["redis_version"]
     arch_bits = info["arch_bits"]
+    cluster_enabled = info["cluster_enabled"]
     REDIS_INFO["version"] = version
     REDIS_INFO["arch_bits"] = arch_bits
+    REDIS_INFO["cluster_enabled"] = cluster_enabled
 
     # module info
     redismod_url = session.config.getoption("--redismod-url")
     info = _get_info(redismod_url)
     REDIS_INFO["modules"] = info["modules"]
 
+    if cluster_enabled:
+        cluster_nodes = session.config.getoption("--cluster-nodes")
+        wait_for_cluster_creation(redis_url, cluster_nodes)
+
+
+def wait_for_cluster_creation(redis_url, cluster_nodes, timeout=20):
+    now = time.time()
+    timeout = now + timeout
```

Comment:
Mislabelled variable. Timeout is the amount of time to wait (i.e in seconds), whereas this is something liek a wait until.... This affects the while loop as well.